src/validation/validator.ts

```diff
@@ -0,0 +1,43 @@
+import Ajv from 'ajv';
+import { schemas } from './json-schemas';
+
+const validator = new Ajv();
+
+for (let schemaName in schemas) {
+  addSchema(schemaName, schemas[schemaName]);
+}
+
+export function addSchema(schemaName: string, schema): void {
+  validator.addSchema(schema, schemaName);
+}
+
+/**
+ * TODO: add JSDoc
+ * @param schemaName
+ * @param payload
+ * @returns
+ */
+export function validate(schemaName: string, payload: any): void {
+  const validateFn = validator.getSchema(schemaName);
+
+  if (!validateFn) {
+    throw new Error(`schema for ${schemaName} not found.`);
+  }
+
+  validateFn(payload);
+
+  if (!validateFn.errors) {
+    return;
+  }
+
+  // AJV is configured by default to stop validating after the 1st error is encountered which means
+  // there will only ever be one error;
+  const [ errorObj ] = validateFn.errors;
+  let { instancePath, message } = errorObj;
```

Comment:
Prefer const over let when possible.
Let's just add the eslint rule and be done with it:
```    'prefer-const': 'error',```
```suggestion
  const { instancePath, message } = errorObj;
```

Reply:
`let` is used here because `instancePath` is re-assigned in the conditional below.

Reply:
Do you see value in enabling linting rule 'prefer-const': 'error'? Or it'd probably annoy you because it'll complain about this exact line due to `message`...

Reply:
definitely see the value in adding it. There's an option that allows re-assignment of destructured properties. added the eslint rule. good call! https://github.com/TBD54566975/dwn-sdk-js/commit/17a1689fc67ff4d02c4dfaed1eea51a34e0a6f85